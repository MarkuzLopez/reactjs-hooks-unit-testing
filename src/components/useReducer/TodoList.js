import React from 'react'
import TodoItem from './TodoItem';


const TodoList = (props) => {
    const { todosArray, toggleTodo, deleteTodo } = props;
    return (
        <ul className="list-group list-group-flush">
        { 
            todosArray.map((todo, index) => (
                <TodoItem
                key={todo.id}
                todo={todo}
                index={index}
                toggleTodo={toggleTodo}
                deleteTodo={deleteTodo}
                />
            ))
        }
        </ul>
     );
}
 
export default TodoList;