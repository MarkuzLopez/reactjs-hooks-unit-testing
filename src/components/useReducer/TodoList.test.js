import { shallow } from 'enzyme'
import React from 'react'
import { demoTodos } from '../../tests/fixtures/demosTodos'
import TodoList from './TodoList'

describe('Pruebas unitaria de componente TodoList', () => {
    
    const toggleTodo = jest.fn();
    const deleteTodo = jest.fn();

    const wrapper = shallow(
        <TodoList 
            todosArray={demoTodos}
            toggleTodo={toggleTodo}
            deleteTodo={deleteTodo}
        />
    );

    test('dene de crear el toMatchSnapshot y mostrarse correctamene ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de tener dos <TodoListItem />', () => {
        
        expect(wrapper.find('TodoItem').length).toBe(demoTodos.length);
        // console.log(wrapper.find('TodoItem').at(0).props()); // para ver todo lo que tiene los props
        expect(wrapper.find('TodoItem').at(0).prop('deleteTodo')).toEqual(expect.any(Function)) // que espero cualquier funcion toEqual();
    });

    
    
})
