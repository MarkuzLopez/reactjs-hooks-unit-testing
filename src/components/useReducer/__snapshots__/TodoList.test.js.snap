// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitaria de componente TodoList dene de crear el toMatchSnapshot y mostrarse correctamene  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoItem
    deleteTodo={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
    toggleTodo={[MockFunction]}
  />
  <TodoItem
    deleteTodo={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Angular",
        "done": false,
        "id": 2,
      }
    }
    toggleTodo={[MockFunction]}
  />
</ul>
`;
